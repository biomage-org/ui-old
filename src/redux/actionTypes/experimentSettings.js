const EXPERIMENT_SETTINGS = 'experimentSettings';

/**
 * Loaded general information about the experiment.
 */
const EXPERIMENT_SETTINGS_INFO_UPDATE = `${EXPERIMENT_SETTINGS}/updateInfo`;

/**
 * Sets the state of the processing component to a state displayable by the system.
 */
const EXPERIMENT_SETTINGS_PROCESSING_CONFIG_LOADED = `${EXPERIMENT_SETTINGS}/loadedProcessingConfig`;

/**
 * Sets error value in store if there is error in loading
 */
const EXPERIMENT_SETTINGS_PROCESSING_ERROR = `${EXPERIMENT_SETTINGS}/errorProcessing`;

/**
 * Updates the processing component's configuration.
 */
const EXPERIMENT_SETTINGS_NON_SAMPLE_FILTER_UPDATE = `${EXPERIMENT_SETTINGS}/updateNonSampleFilterSettings`;

/**
 * Updates the processing component's configuration.
 */
const EXPERIMENT_SETTINGS_SAMPLE_FILTER_UPDATE = `${EXPERIMENT_SETTINGS}/updateSampleFilterSettings`;

/**
 * Saves the current state of the configuration to DynamoDB.
 */
const EXPERIMENT_SETTINGS_PROCESSING_SAVE = `${EXPERIMENT_SETTINGS}/saveProcessing`;

/**
 * Signals that a pipeline run was initiated by the user.
 */
const EXPERIMENT_SETTINGS_QC_START = `${EXPERIMENT_SETTINGS}/startQC`;

/**
 * A QC step was enabled or disabled
 */
const EXPERIMENT_SETTINGS_SET_QC_STEP_ENABLED = `${EXPERIMENT_SETTINGS}/setQCStepEnabled`;

/**
 * The filter settings for one sample in one step were copied over to all other samples
 */
const EXPERIMENT_SETTINGS_COPY_SETTINGS_TO_ALL_SAMPLES = `${EXPERIMENT_SETTINGS}/copyFilterSettingsToAllSamples`;

const EXPERIMENT_SETTINGS_SET_SAMPLE_FILTER_SETTINGS_AUTO = `${EXPERIMENT_SETTINGS}/setSampleFilterSettingsAuto`;

/**
 * The filter settings for a filter were changed
 */
const EXPERIMENT_SETTINGS_ADD_CHANGED_QC_FILTER = `${EXPERIMENT_SETTINGS}/addChangedQCFilter`;

/**
 * The pending settings for filters were discarded
 */
const EXPERIMENT_SETTINGS_DISCARD_CHANGED_QC_FILTERS = `${EXPERIMENT_SETTINGS}/discardChangedQCFilters`;

/**
 * Received an update from qc for a sample in a step
 */
const EXPERIMENT_SETTINGS_UPDATE_SAMPLE_FROM_QC = `${EXPERIMENT_SETTINGS}/updateSampleProcessingSettingsFromQC`;

const EXPERIMENT_SETTINGS_PIPELINE_VERSION_UPDATED = `${EXPERIMENT_SETTINGS}/pipelineVersionUpdated`;

export {
  EXPERIMENT_SETTINGS_INFO_UPDATE,
  EXPERIMENT_SETTINGS_PROCESSING_CONFIG_LOADED,
  EXPERIMENT_SETTINGS_NON_SAMPLE_FILTER_UPDATE,
  EXPERIMENT_SETTINGS_SAMPLE_FILTER_UPDATE,
  EXPERIMENT_SETTINGS_PROCESSING_SAVE,
  EXPERIMENT_SETTINGS_PROCESSING_ERROR,
  EXPERIMENT_SETTINGS_QC_START,
  EXPERIMENT_SETTINGS_SET_QC_STEP_ENABLED,
  EXPERIMENT_SETTINGS_COPY_SETTINGS_TO_ALL_SAMPLES,
  EXPERIMENT_SETTINGS_SET_SAMPLE_FILTER_SETTINGS_AUTO,
  EXPERIMENT_SETTINGS_ADD_CHANGED_QC_FILTER,
  EXPERIMENT_SETTINGS_DISCARD_CHANGED_QC_FILTERS,
  EXPERIMENT_SETTINGS_UPDATE_SAMPLE_FROM_QC,
  EXPERIMENT_SETTINGS_PIPELINE_VERSION_UPDATED,
};
