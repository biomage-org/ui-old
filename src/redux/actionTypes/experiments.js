const EXPERIMENTS = 'experiments';

/**
 * Create experiments
 */
const EXPERIMENTS_CREATED = `${EXPERIMENTS}/created`;

/**
 * Create experiments
 */
const EXPERIMENTS_LOADING = `${EXPERIMENTS}/loading`;

/**
 * Experiments loaded
 */
const EXPERIMENTS_LOADED = `${EXPERIMENTS}/loaded`;

/**
 * Example experiments loaded
 */
const EXPERIMENTS_EXAMPLES_LOADED = `${EXPERIMENTS}/examplesLoaded`;

/**
 * Experiments updated
 */
const EXPERIMENTS_UPDATED = `${EXPERIMENTS}/updated`;

/**
 * Delete experiments
 */
const EXPERIMENTS_DELETED = `${EXPERIMENTS}/deleted`;

/**
 * Saving experiment
 */
const EXPERIMENTS_SAVING = `${EXPERIMENTS}/saving`;

const EXPERIMENTS_SAVED = `${EXPERIMENTS}/saved`;

/**
 * Update experiments
 */
const EXPERIMENTS_ERROR = `${EXPERIMENTS}/error`;

const EXPERIMENTS_SWITCH = `${EXPERIMENTS}/switch`;

/**
 * Set active project.
 */
const EXPERIMENTS_SET_ACTIVE = `${EXPERIMENTS}/setActive`;

/**
 * Add project metadata.
 */
const EXPERIMENTS_METADATA_CREATE = `${EXPERIMENTS}/metadataCreate`;

/**
 * Update project metadata.
 */
const EXPERIMENTS_METADATA_RENAME = `${EXPERIMENTS}/metadataRename`;

/**
 * Delete project metadta.
 */
const EXPERIMENTS_METADATA_DELETE = `${EXPERIMENTS}/metadataDelete`;

export {
  EXPERIMENTS_CREATED,
  EXPERIMENTS_UPDATED,
  EXPERIMENTS_ERROR,
  EXPERIMENTS_LOADING,
  EXPERIMENTS_LOADED,
  EXPERIMENTS_EXAMPLES_LOADED,
  EXPERIMENTS_DELETED,
  EXPERIMENTS_SAVING,
  EXPERIMENTS_SAVED,
  EXPERIMENTS_SWITCH,
  EXPERIMENTS_SET_ACTIVE,
  EXPERIMENTS_METADATA_CREATE,
  EXPERIMENTS_METADATA_RENAME,
  EXPERIMENTS_METADATA_DELETE,
};
