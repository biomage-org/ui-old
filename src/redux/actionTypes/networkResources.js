const NETWORK_RESOURCES = 'networkResources';

/**
 * Loads name environment for the current request.
 */
const NETWORK_RESOURCES_DEPLOYMENT_INFO_LOADED = `${NETWORK_RESOURCES}/deploymentInfoLoaded`;

export {
  // eslint-disable-next-line import/prefer-default-export
  NETWORK_RESOURCES_DEPLOYMENT_INFO_LOADED,
};
