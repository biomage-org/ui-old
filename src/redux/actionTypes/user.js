const USER = 'user';

/**
 * User was loaded
 */
const USER_LOADED = `${USER}/loaded`;

export {
  // eslint-disable-next-line import/prefer-default-export
  USER_LOADED,
};
