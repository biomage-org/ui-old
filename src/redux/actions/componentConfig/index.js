import loadPlotConfig from 'redux/actions/componentConfig/loadPlotConfig';
import loadComponentConfig from 'redux/actions/componentConfig/loadComponentConfig';
import updatePlotConfig from 'redux/actions/componentConfig/updatePlotConfig';
import savePlotConfig from 'redux/actions/componentConfig/savePlotConfig';
import getDotPlot from 'redux/actions/componentConfig/getDotPlot';
import updatePlotData from 'redux/actions/componentConfig/updatePlotData';
import resetPlotConfig from 'redux/actions/componentConfig/resetPlotConfig';

export {
  loadPlotConfig,
  loadComponentConfig,
  updatePlotConfig,
  savePlotConfig,
  getDotPlot,
  updatePlotData,
  resetPlotConfig,
};
