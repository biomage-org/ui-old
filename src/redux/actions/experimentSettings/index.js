import saveProcessingSettings from './processingConfig/saveProcessingSettings';
import loadProcessingSettings from './processingConfig/loadProcessingSettings';
import loadedProcessingConfig from './processingConfig/loadedProcessingConfig';
import setQCStepEnabled from './processingConfig/setQCStepEnabled';
import copyFilterSettingsToAllSamples from './processingConfig/copyFilterSettingsToAllSamples';
import setSampleFilterSettingsAuto from './processingConfig/setSampleFilterSettingsAuto';
import updateFilterSettings from './processingConfig/updateFilterSettings';
import addChangedQCFilter from './processingConfig/addChangedQCFilter';
import discardChangedQCFilters from './processingConfig/discardChangedQCFilters';
import updateProcessingSettingsFromQC from './processingConfig/updateProcessingSettingsFromQC';
import updateExperimentInfo from './updateExperimentInfo';
import updatePipelineVersion from './updatePipelineVersion';

export {
  loadProcessingSettings,
  loadedProcessingConfig,
  saveProcessingSettings,
  updateExperimentInfo,
  setQCStepEnabled,
  copyFilterSettingsToAllSamples,
  setSampleFilterSettingsAuto,
  addChangedQCFilter,
  discardChangedQCFilters,
  updateFilterSettings,
  updateProcessingSettingsFromQC,
  updatePipelineVersion,
};
