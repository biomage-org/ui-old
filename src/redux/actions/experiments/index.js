import createExperiment from './createExperiment';
import deleteExperiment from './deleteExperiment';
import updateExperiment from './updateExperiment';
import loadExperiments from './loadExperiments';
import setActiveExperiment from './setActiveExperiment';
import switchExperiment from './switchExperiment';
import reorderSamples from './reorderSamples';
import createMetadataTrack from './createMetadataTrack';
import renameMetadataTrack from './renameMetadataTrack';
import deleteMetadataTrack from './deleteMetadataTrack';
import updateValueInMetadataTrack from './updateValueInMetadataTrack';
import cloneExperiment from './cloneExperiment';
import reloadExperimentInfo from './reloadExperimentInfo';

export {
  createExperiment,
  deleteExperiment,
  updateExperiment,
  loadExperiments,
  setActiveExperiment,
  switchExperiment,
  reorderSamples,
  createMetadataTrack,
  renameMetadataTrack,
  deleteMetadataTrack,
  updateValueInMetadataTrack,
  cloneExperiment,
  reloadExperimentInfo,
};
