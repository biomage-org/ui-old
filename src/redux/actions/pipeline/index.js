import runQC from './runQC';
import runGem2s from './runGem2s';
import runSubsetExperiment from './runSubsetExperiment';

export {
  runQC,
  runGem2s,
  runSubsetExperiment,
};
