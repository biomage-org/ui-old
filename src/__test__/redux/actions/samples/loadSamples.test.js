import _ from 'lodash';

import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import fetchMock, { enableFetchMocks } from 'jest-fetch-mock';

import initialSampleState from 'redux/reducers/samples/initialState';
import { SAMPLES_ERROR, SAMPLES_LOADED, SAMPLES_LOADING } from 'redux/actionTypes/samples';
import { loadSamples } from 'redux/actions/samples';

enableFetchMocks();

const mockStore = configureStore([thunk]);

describe('loadSample action', () => {
  const experimentId = '1234';

  const initialState = {
    samples: {
      ...initialSampleState,
    },
  };

  const response = new Response(
    JSON.stringify(
      [
        {
          samples: {
            ids: ['sample-1', 'sample-2'],
            'sample-1': { name: 'sample-1' },
            'sample-2': { name: 'sample-2' },
          },
        }],
    ),
  );

  beforeEach(() => {
    fetchMock.resetMocks();
    fetchMock.doMock();
    fetchMock.mockResolvedValue(response);
  });

  it('Works correctly', async () => {
    // mock new response from apiv2
    const responseV2 = new Response(
      JSON.stringify(
        [{
          id: 'e03ef6ea-5014-4e57-aecd-59964ac9172c',
          experimentId: 'fd8d5f24-a2d9-da28-2ffd-725b99b1127b',
          name: 'BLp7',
          sampleTechnology: '10x',
          createdAt: '2021-12-07 17:36:27.773+00',
          updatedAt: '2021-12-07 17:38:42.036+00',
          metadata: { age: 'BL', timePoint: 'BL' },
          files: {
            matrix10X: { uploadStatus: 'uploaded', sampleFileType: 'matrix10x' },
            barcodes10X: { uploadStatus: 'uploaded', sampleFileType: 'barcodes10x' },
            features10X: { uploadStatus: 'uploaded', sampleFileType: 'features10x' },
          },
          options: {},
        }],

      ),
    );
    fetchMock.resetMocks();
    fetchMock.doMock();
    fetchMock.mockResolvedValue(responseV2);

    const store = mockStore(initialState);
    await store.dispatch(loadSamples(experimentId));

    const action1 = store.getActions()[1];
    expect(action1.type).toEqual(SAMPLES_LOADED);
    expect(action1.payload).toMatchSnapshot();
  });

  it('Dispatches error correctly', async () => {
    fetchMock.resetMocks();
    fetchMock.doMock();
    fetchMock.mockRejectOnce(new Error('An api error'));

    const store = mockStore(initialState);
    await store.dispatch(loadSamples(experimentId));

    const actions = store.getActions();
    expect(_.map(actions, 'type')).toEqual([SAMPLES_LOADING, SAMPLES_ERROR]);
    expect(_.map(actions, 'payload')).toMatchSnapshot();
  });
});
