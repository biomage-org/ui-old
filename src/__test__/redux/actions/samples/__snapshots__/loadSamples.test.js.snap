// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadSample action Dispatches error correctly 1`] = `
Array [
  undefined,
  Object {
    "error": "Error: An api error",
  },
]
`;

exports[`loadSample action Works correctly 1`] = `
Object {
  "samples": Object {
    "e03ef6ea-5014-4e57-aecd-59964ac9172c": Object {
      "createdDate": "2021-12-07 17:36:27.773+00",
      "experimentId": "1234",
      "fileNames": Array [
        "matrix.mtx.gz",
        "barcodes.tsv.gz",
        "features.tsv.gz",
      ],
      "files": Object {
        "barcodes.tsv.gz": Object {
          "name": "barcodes.tsv.gz",
          "size": undefined,
          "upload": Object {
            "status": "uploaded",
          },
          "valid": true,
        },
        "features.tsv.gz": Object {
          "name": "features.tsv.gz",
          "size": undefined,
          "upload": Object {
            "status": "uploaded",
          },
          "valid": true,
        },
        "matrix.mtx.gz": Object {
          "name": "matrix.mtx.gz",
          "size": undefined,
          "upload": Object {
            "status": "uploaded",
          },
          "valid": true,
        },
      },
      "lastModified": "2021-12-07 17:38:42.036+00",
      "metadata": Object {
        "age": "BL",
        "timePoint": "BL",
      },
      "name": "BLp7",
      "options": Object {},
      "type": "10x",
      "uuid": "e03ef6ea-5014-4e57-aecd-59964ac9172c",
    },
  },
}
`;
